fastly/provider.go

```diff
@@ -35,6 +35,7 @@ func Provider() terraform.ResourceProvider {
 			"fastly_service_dynamic_snippet_content_v1": resourceServiceDynamicSnippetContentV1(),
 			"fastly_user_v1":                            resourceUserV1(),
 			"fastly_service_waf_configuration":          resourceServiceWAFConfigurationV1(),
+			"fastly_tls_private_key":                    resourceTLSPrivateKeyV1(),
```

Comment:
The pattern for the naming of these functions would be the name of the resource without the provider prefix - 'fastly' in this case - in camel case rather than snake. This should be just `resourceTLSPrivateKey`

Reply:
So you mean skip the pattern of sticking the `V1` at the end?
```suggestion
			"fastly_tls_private_key":                    resourceTLSPrivateKey(),
```

Reply:
Yep - example: https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/provider.go#L404. The only examples I can think which break this pattern is this, due to how `aws_lb` and `aws_alb` are interchangeable: https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/provider.go#L1062-L1063

Reply:
Ok thanks, I'll push that in a bit. Out of interest is there a reason the other resources have the V1 on them or is that just a pattern we're trying to move away from?